### PR TITLE
Run unit tests using built wheel

### PR DESCRIPTION
**Pull request by vowelparrot**
For context, I inadvertently used pexpect as a dependency (fixed in https://github.com/hwchase17/langchain/commit/1b5721c999c9fc310cefec383666f43c80ec9620)

This slipped through CI because pexpect is a `poetry` dependency, making it a sub-dependency of our CI system.

The proposal is a workaround to install the poetry dependencies with pip without directly relying on poetry during unit testing.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -74,6 +74,17 @@ azure-cosmos = {version="^4.4.0b1", optional=true}
 lark = {version="^1.1.5", optional=true}
 lancedb = {version = "^0.1", optional = true}
 pexpect = {version = "^4.8.0", optional = true}
+# Used for CI testing
```
Comment (vowelparrot) [inline pyproject.toml:77]: These won't be exported unless they're added here for some reason. Would be open to suggestions from any non-poetry novice

File: tests/integration_tests/prompts/test_jinja2.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
+"""Test functionality related to prompts."""
+from typing import Dict, List, Tuple
+
+import pytest
+
+from langchain.prompts.few_shot import FewShotPromptTemplate
+from langchain.prompts.prompt import PromptTemplate
+
+
+def test_prompt_from_jinja2_template() -> None:
```
Comment (vowelparrot) [inline tests/integration_tests/prompts/test_jinja2.py:10]: These were using optional dependencies (jinja2). 

I'm not sure if we want to have something that runs more frequently than integration tests to test this?

**Review by eyurtsev — APPROVED** [review]

File: tests/integration_tests/prompts/test_jinja2.py (hunk 2 above)
Comment (eyurtsev) [reply tests/integration_tests/prompts/test_jinja2.py:10]: Was this picked up from poetry as well?

Comment (eyurtsev) [reply tests/integration_tests/prompts/test_jinja2.py:10]: oh nevermind -- misunderstood your comment

**Pull request closed without merging**